service/controller/cloudconfig/cloud_config.go

```diff
@@ -23,6 +23,39 @@ const (
 iscsid.safe_logout = Yes`
 )
 
+const calicoKubeKillScript = `#!/bin/bash
+
+HOSTNAME=$(hostname | tr '[:upper:]' '[:lower:]')
+
+while [ "$(kubectl get nodes $HOSTNAME -o jsonpath='{.metadata.name}')" != "$HOSTNAME" ]; do
+  sleep 1
+  echo "Waiting for node $HOSTNAME to be registered"
+done
+
+sleep 30s
+
+RETRY=5
+result=""
+
+while [ "$result" != "ok" ] && [ $RETRY -gt 0 ]; do
+  sleep 10s
+  echo "Trying to restart k8s services ..."
+  let RETRY=$RETRY-1
+  kubectl -n kube-system delete pod -l k8s-app=calico-node && \
+    sleep 1m && \
```

Comment:
Does `kubectl wait` not work here?

Reply:
Yes - this code was just copied from the previous location by thomas. Let's sort this out in a separate PR.
Line should be `kubectl wait -n kube-system -l k8s-app=calico-node --for=delete --timeout=120s` or similar.